src/dbnode/storage/shard.go

```diff
@@ -1823,54 +1910,38 @@ func (s *dbShard) Bootstrap(
 			dbBlocks.Tags.Finalize()
 		}
 
-		// Cannot close blocks once done as series takes ref to these
-		bsResult, err := entry.Series.Bootstrap(dbBlocks.Blocks, blockStates)
-		if err != nil {
-			multiErr = multiErr.Add(err)
+		if bootstrap {
+			bsResult, err := entry.Series.Bootstrap(dbBlocks.Blocks, blockStatesSnapshot)
+			if err != nil {
+				multiErr = multiErr.Add(err)
+			}
+			shardBootstrapResult.update(bsResult)
+		} else {
+			entry.Series.Load(dbBlocks.Blocks, blockStatesSnapshot)
```

Comment:
Instead of having two separate APIs at the `series` package, maybe just a single API `Load(...)` with `LoadOptions struct { Bootstrap bool; }` and a `return LoadResult{BootstrapResult: bsResult}` from the method?

Reply:
I really don't like using booleans like that in public APis, much prefer to have two separate APIs and keep boolean switches like that in internal methods personally (I understand it looks like API bloat but having an optional bool is basically the same API bloat but just in a round about way)